src/transformers/modeling_utils.py

```diff
@@ -2159,7 +2185,7 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
         model.eval()
 
         # Dispatch model with hooks on all devices if necessary
-        if device_map is not None:
+        if device_map:
```

Comment:
Please revert, we always explicitly test instead of relying on Python bool magic.

Reply:
Great !  Should be fixed in a4c19c193b598efae80f2489298c3aeeacb50804